README.md

```diff
@@ -406,6 +406,16 @@ Then curl the metrics endpoint
 
 To run the e2e tests locally see the documentation in [tests/README.md](./tests/README.md).
 
+##### Debug CustomStateMetrics feature
```

Comment:
Instead of adding it to the README, I think it would be better to put it in its own doc under https://github.com/kubernetes/kube-state-metrics/tree/main/docs, or maybe in a new directory there.

I suggested reworking the structure of our doc directory in the past, we might want to revive that effort.